Failed to build with prost 0.10

**Describe the bug**
Attempting to use `grpcio 0.10` + `grpcio-compiler 0.10` with `prost 0.10` fails to build.
IIUC it's because of `grpcio`'s dependency on `prost 0.9`

**To Reproduce**
Steps to reproduce the behavior:
1. Make a new crate and add the following deps:
    ```toml
    [dependencies]
    grpcio = { version = "0.10", default-features = false, features = ["prost-codec"] }
    prost = "0.10"

    [build-dependencies]
    grpcio-compiler = { version = "0.10", default-features = false, features = ["prost-codec"] }
    ```
1. Add a `.proto` file and a build script that generates Rust code via `grpcio_compiler::prost_codegen::compile_protos`
1. Observe error:
```
error[E0277]: the trait bound `HealthCheckResponse: prost::message::Message` is not satisfied
  --> .../out/protos-auto-gen/grpc.health.v1.rs:40:18
   |
40 |             ser: ::grpcio::pr_ser,
   |                  ^^^^^^^^^^^^^^^^ the trait `prost::message::Message` is not implemented for `HealthCheckResponse`
   |
   = help: the following other types implement trait `prost::message::Message`:
             ()
             Box<M>
             Vec<u8>
             bool
             f32
             f64
             i32
             i64
           and 4 others
note: required by a bound in `pr_ser`
  --> /home/remoun/.cargo/registry/src/github.com-1ecc6299db9ec823/grpcio-0.10.3/src/codec.rs:76:19
   |
76 |     pub fn ser<M: Message>(msg: &M, buf: &mut GrpcSlice) -> Result<()> {
   |                   ^^^^^^^ required by this bound in `pr_ser`

error[E0277]: the trait bound `HealthCheckResponse: prost::message::Message` is not satisfied
  --> .../out/protos-auto-gen/grpc.health.v1.rs:41:17
   |
41 |             de: ::grpcio::pr_de,
   |                 ^^^^^^^^^^^^^^^ the trait `prost::message::Message` is not implemented for `HealthCheckResponse`
   |
   = help: the following other types implement trait `prost::message::Message`:
             ()
             Box<M>
             Vec<u8>
             bool
             f32
             f64
             i32
             i64
           and 4 others
note: required by a bound in `pr_de`
  --> /home/remoun/.cargo/registry/src/github.com-1ecc6299db9ec823/grpcio-0.10.3/src/codec.rs:94:18
   |
94 |     pub fn de<M: Message + Default>(mut reader: MessageReader) -> Result<M> {
   |                  ^^^^^^^ required by this bound in `pr_de`
```

**Expected behavior**
Build succeeds.

**System information**
* CPU architecture: x86_64
* Distribution and kernel version: `uname -a` output: `Linux remoun-mc 5.10.102.1-microsoft-standard-WSL2 #1 SMP Wed Mar 2 00:30:59 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux`
* SELinux on?: I don't think so
* Any other system details we should know?: I'm running WSL2, but I expect that doesn't matter here.

**Additional context**
I verified the generated struct includes `derive(Message)`:
```rust
#[derive(Clone, PartialEq, ::prost::Message)]
pub struct HealthCheckResponse {
    #[prost(enumeration = "health_check_response::ServingStatus", tag = "1")]
    pub status: i32,
}
```

So I'm pretty sure it's because of `grpcio`'s dependency on `prost 0.9`

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -22,7 +22,7 @@ libc = "0.2"
 futures-executor = "0.3"
 futures-util = { version = "0.3", default-features = false, features = ["std", "sink"] }
 protobuf = { version = "2.0", optional = true }
-prost = { version = "0.9", optional = true }
+prost = { version = "0.10", optional = true }
 bytes = { version = "1.0", optional = true }
 log = "0.4"
 parking_lot = "0.11"

--- a/compiler/Cargo.toml
+++ b/compiler/Cargo.toml
@@ -18,9 +18,9 @@ prost-codec = ["prost-build", "prost-types", "prost", "derive-new", "tempfile"]
 
 [dependencies]
 protobuf = { version = "2", optional = true }
-prost = { version = "0.9", optional = true }
-prost-build = { version = "0.9", optional = true }
-prost-types = { version = "0.9", optional = true }
+prost = { version = "0.10", optional = true }
+prost-build = { version = "0.10", optional = true }
+prost-types = { version = "0.10", optional = true }
 derive-new = { version = "0.5", optional = true }
 tempfile = { version = "3.0", optional = true }
 

--- a/health/Cargo.toml
+++ b/health/Cargo.toml
@@ -21,6 +21,6 @@ prost-codec = ["grpcio/prost-codec", "prost"]
 futures-executor = "0.3"
 futures-util = { version = "0.3", default-features = false, features = ["std"] }
 grpcio = { path = "..", version = "0.10.0", default-features = false }
-prost = { version = "0.9", optional = true }
+prost = { version = "0.10", optional = true }
 protobuf = { version = "2", optional = true }
 log = "0.4"

--- a/proto/Cargo.toml
+++ b/proto/Cargo.toml
@@ -20,9 +20,9 @@ prost-codec = ["prost-derive", "prost-types", "bytes", "lazy_static", "grpcio/pr
 [dependencies]
 grpcio = { path = "..", features = ["boringssl"], version = "0.10.0", default-features = false }
 bytes = { version = "1.0", optional = true }
-prost = { version = "0.9", optional = true }
-prost-derive = { version = "0.9", optional = true }
-prost-types = { version = "0.9", optional = true }
+prost = { version = "0.10", optional = true }
+prost-derive = { version = "0.10", optional = true }
+prost-types = { version = "0.10", optional = true }
 protobuf = "2"
 lazy_static = { version = "1.3", optional = true }
 

--- a/tests-and-examples/Cargo.toml
+++ b/tests-and-examples/Cargo.toml
@@ -18,7 +18,7 @@ futures-executor = "0.3"
 futures-util = { version = "0.3", features = ["sink"] }
 futures-timer = "3.0"
 protobuf = { version = "2.22", optional = true }
-prost = { version = "0.9", optional = true }
+prost = { version = "0.10", optional = true }
 bytes = { version = "1.0", optional = true }
 log = "0.4"
 grpcio = { path = "..", default-features = false, features = ["boringssl"] }

